package/iframe/js/features.js

```diff
@@ -5,30 +5,62 @@ var logger = require('./logger');
-        // ['hardware.memory', 512],  // getFeature() with comparison.
-        // 'api.window.MozMobileNetworkInfo',  // hasFeature().
+// See zamboni docs for the order - it matters, we'll push promises in the same
+// order to generate the features signature.
+var FEATURES = [
+    // false,  // Hardcoded boolean value.
+    // 'mozApps' in navigator,  // Dynamic boolean value.
+    // ['hardware.memory', 512],  // getFeature() with comparison.
+    // 'api.window.MozMobileNetworkInfo',  // hasFeature().
+
+    // We are only interested in a few features for now. We're already only
+    // doing this if getFeature() is present, and it was introduced in 2.0, so
+    // we know we can hardcode anything that comes with 2.0 and for which we
+    // don't need to know the exact value.
+    true, // 'getMobileIdAssertion' in window.navigator || 'api.window.Navigator.getMobileIdAssertion',
+    true, // 'manifest.precompile',
+    ['hardware.memory', 512],
+    ['hardware.memory', 1024],
+    true, // NFC
+    'acl.version', // OpenMobile ACL
+];
+
+function FeaturesBitField(size, values) {
+    /**
+     * Class that stores the bits into several 8-bit unsigned integers, and 
+     * can import/export from/to base64. Designed that way to be compatible
+     * with the way we read the features signature in Python.
+    **/
+    this.size = size;
+    this.values = new Uint8Array(Math.ceil(this.size / 8));
+}
 
-        // We are only interested in a few features for now. We're already
-        // only doing this if getFeature() is present, and it was
-        // introduced in 2.0, so we know we can hardcode anything that
-        // comes with 2.0 and for which we don't need to know the exact
-        // value.
-        true, // 'getMobileIdAssertion' in window.navigator || 'api.window.Navigator.getMobileIdAssertion',
-        true, // 'manifest.precompile',
-        ['hardware.memory', 512],
-        ['hardware.memory', 1024],
-        true, // NFC
-        'acl.version', // OpenMobile ACL
-        // Don't add any more as long as bug 1172487 is not fixed, it won't
-        // work correctly.
-    ];
+FeaturesBitField.prototype.get = function(i) {
+    var index = (i / 8) | 0;
+    var bit = i % 8;
+    return (this.values[index] & (1 << bit)) !== 0;
+};
+
+FeaturesBitField.prototype.set = function(i, value) {
+    var index = (i / 8) | 0;
+    var bit = i % 8;
+    if (value) {
+        this.values[index] |= 1 << bit;
+    } else {
+        this.values[index] &= ~(1 << bit);
+    }
+};
+
+FeaturesBitField.prototype.toBase64 = function() {
+    return btoa(String.fromCharCode.apply(null, this.values));
+};
+
+function buildFeaturesPromises(features, navigator) {
+    navigator = navigator || window.navigator;
+    var promises = [];
+    // Execute getFeature('hardware.memory') immediately and store the promise
+    // to be able to re-use it without making the call twice.
+    var memoryPromise = navigator.getFeature('hardware.memory');
```

Comment:
I didn't want to do something too generic for now because we only have one `getFeature()` call to cache right now.

(btw, `getFeature('hardware.memory')` takes anywhere between 60 and 160 ms on my flame, this is why I cached it)